core/src/main/java/org/mule/transport/polling/watermark/selector/MaxValueWatermarkSelector.java

```diff
@@ -17,9 +17,9 @@
 public class MaxValueWatermarkSelector extends ComparableWatermarkSelector
 {
 
-    @Override
-    protected int comparableQualifier()
-    {
-        return 1;
-    }
+	@Override
+	public boolean compare(Comparable<Object> value, Comparable<Object> currentWatermark) {
```

Comment:
a method named compare which operates over two Comparable objects shouldn't return boolean because it would be impossible to build a selector which considers equal values.